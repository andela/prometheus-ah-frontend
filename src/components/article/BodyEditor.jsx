import React, { Component } from 'react';
import { Editor } from '@tinymce/tinymce-react';
import PropTypes from 'prop-types';

/**
 * @class BodyEditor
 */
class BodyEditor extends Component {
  /**
   *@returns {*} jsx
   */
  render() {
    const {
      handleChange, value, handleBodyFocusIn, handleBodyFocusOut
    } = this.props;
    return (
      <Editor
        apiKey="zre5x72iyosuc7o3z1vg6lfbkxwdw4b5lr9yrwpgjsnfsca7"
        onEditorChange={handleChange}
        value={value}
        onFocusIn={handleBodyFocusIn}
        onFocusOut={handleBodyFocusOut}
        init={{
          menubar: false,
          theme: 'inlite',
          inline: true,
          mobile: {
            theme: 'mobile',
            plugins: ['autosave', 'lists', 'autolink']
          },
          plugins: [
            'autolink',
            'codesample',
            'link',
            'linkchecker',
            'lists',
            'mediaembed',
            'textcolor',
            'image',
            'emoticons',
          ],
          selection_toolbar:
            `bold italic underline strikethrough|
            h2 h3 | blockquote quicklink | alignleft aligncenter alignright alignjustify |
            forecolor fontsize underlinr`,
          images_reuse_filename: true,
          toolbar: 'emoticons'
        }}
      />
    );
  }
}
BodyEditor.propTypes = {
  handleChange: PropTypes.func,
  value: PropTypes.string,
  handleBodyFocusIn: PropTypes.func,
  handleBodyFocusOut: PropTypes.func,

};

export default BodyEditor;
