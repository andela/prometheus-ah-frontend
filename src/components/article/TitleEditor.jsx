import React, { Component } from 'react';
import { Editor } from '@tinymce/tinymce-react';
import PropTypes from 'prop-types';


/**
 * @class BodyEditor
 */
class TitleEditor extends Component {
  /**
   *@returns {*} jsx
   */
  render() {
    const {
      handleChange, value, handleTitleFocusIn, handleTitleFocusOut,
    } = this.props;
    return (
      <Editor
        apiKey="zre5x72iyosuc7o3z1vg6lfbkxwdw4b5lr9yrwpgjsnfsca7"
        onEditorChange={handleChange}
        value={value}
        onFocusIn={handleTitleFocusIn}
        onFocusOut={handleTitleFocusOut}
        name="title-editor"
        tagName="h1"
        init={{
          menubar: false,
          inline: true,
          theme: 'inlite',
          insert_toolbar: 'undo redo',
          selection_toolbar: 'italic underline',
        }}
      />
    );
  }
}

TitleEditor.propTypes = {
  handleChange: PropTypes.func,
  value: PropTypes.string,
  handleTitleFocusIn: PropTypes.func,
  handleTitleFocusOut: PropTypes.func,
};
export default TitleEditor;
