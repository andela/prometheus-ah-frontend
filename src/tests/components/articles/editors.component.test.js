import React from 'react';
import { shallow } from 'enzyme';
import BodyEditor from '../../../components/article/BodyEditor';
import TitleEditor from '../../../components/article/TitleEditor';


describe('Editor component', () => {
  it('should render Body Editor correctly', () => {
    const wrapper = shallow(<BodyEditor />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Title Editor correctly', () => {
    const wrapper = shallow(<TitleEditor />);
    expect(wrapper).toMatchSnapshot();
  });
});
