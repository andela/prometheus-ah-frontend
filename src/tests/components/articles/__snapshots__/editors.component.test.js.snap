// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor component should render Body Editor correctly 1`] = `
<Editor
  apiKey="zre5x72iyosuc7o3z1vg6lfbkxwdw4b5lr9yrwpgjsnfsca7"
  init={
    Object {
      "images_reuse_filename": true,
      "inline": true,
      "menubar": false,
      "mobile": Object {
        "plugins": Array [
          "autosave",
          "lists",
          "autolink",
        ],
        "theme": "mobile",
      },
      "plugins": Array [
        "autolink",
        "codesample",
        "link",
        "linkchecker",
        "lists",
        "mediaembed",
        "textcolor",
        "image",
        "emoticons",
      ],
      "selection_toolbar": "bold italic underline strikethrough|
            h2 h3 | blockquote quicklink | alignleft aligncenter alignright alignjustify |
            forecolor fontsize underlinr",
      "theme": "inlite",
      "toolbar": "emoticons",
    }
  }
/>
`;

exports[`Editor component should render Title Editor correctly 1`] = `
<Editor
  apiKey="zre5x72iyosuc7o3z1vg6lfbkxwdw4b5lr9yrwpgjsnfsca7"
  init={
    Object {
      "inline": true,
      "insert_toolbar": "undo redo",
      "menubar": false,
      "selection_toolbar": "italic underline",
      "theme": "inlite",
    }
  }
  name="title-editor"
  tagName="h1"
/>
`;
